frontend/packages/dev-console/src/components/project-access/projectAccessForm-utils.ts

```diff
@@ -0,0 +1,78 @@
+import * as yup from 'yup';
+import * as _ from 'lodash';
+
+export interface UserRole {
+  metadata: {
+    name: string;
+  };
+  roleRef: {
+    name: string;
+  };
+  subjects: [
+    {
+      name: string;
+    }
+  ];
+}
+
+export interface UserRoleBinding {
+  roleBindingName?: string;
+  user: string;
+  role: string;
+}
+
+export const validationSchema = yup.object().shape({
+  projectAccess: yup.array().of(
+    yup.object().shape({
+      user: yup.string().required('Required'),
+      role: yup.string().required('Required'),
+    }),
+  ),
+});
+
+export const getNewRoles = (initialRoles: UserRoleBinding[], formValues: UserRoleBinding[]) => {
```

Comment:
Missing unit tests for utilities.

Reply:
@christianvogt  I have tests for the utilities in : /project-access/__tests__/ProjectAccess.spec.tsx

Reply:
The spec should be named the same as the file it is testing to ensure there's a 1:1 relationship for unit tests.